### PR TITLE
Create OSSF scorecard

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@superm1 this gets us a scorecard like https://securityscorecards.dev/viewer/?uri=github.com/hughsie/libxmlb

**Comment by superm1** [comment]
Sounds good, but can we add a badge or link in README then?

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```
Comment (superm1) [inline .github/workflows/scorecard.yml:35]: is it not possible to use a tag instead?

Comment (hughsie) [reply .github/workflows/scorecard.yml:35]: It's the example file -- I guess there was a good reason to use a commit

**Comment by hughsie** [comment]
> Sounds good, but can we add a badge or link in README then?

Good idea, done.

**Review by superm1 — APPROVED** [review]

**Pull request merged**